lib/policy_machine.rb

```diff
@@ -194,17 +207,23 @@ def privileges
   #
   # TODO:  might make privilege a class of its own
   def scoped_privileges(user_or_attribute, object_or_attribute, options = {})
-    privileges_and_prohibitions = if policy_machine_storage_adapter.respond_to?(:scoped_privileges)
-      policy_machine_storage_adapter.scoped_privileges(user_or_attribute.stored_pe, object_or_attribute.stored_pe, options).map do |op|
+    options = options.dup
+    filters = options.delete(:filters)
```

Comment:
Using `dup` and `delete` here seems strange to me. Since `filters` is only used once, I'd be find with just calling `options[:filters]` below instead of using `filters`. If you want a separate variable, then I think this would be more standard:
```ruby
filters = options[:filters]
```